xrayvision/Visibillity.py

```diff
@@ -44,6 +45,22 @@ def from_map(self, inmap):
         """
         return Transform.dft(inmap, self.uv, self.vis)
 
+    def from_sunpy_map(self, inmap: sunpy.map):
+        """
+        Calculates the visibilities for the preset UV coordinates
+        from a sunpy map
+
+        Parameters
+        ----------
+        inmap
+
+        Returns
+        -------
+
+        """
+        Transform.dft(inmap.data, self.uv, self.vis)
```

Comment:
These need take into account the the additional information a map contains:
* location of the centre in coordinate space
* pixel size
Then this needs to use the to use the dft_map and idft_map methods which need to be updated to handle maps which are not centred on 0, 0